packages/mdc-toolbar/foundation.ts

```diff
@@ -37,65 +74,63 @@ export default class MDCToolbarFoundation extends MDCFoundation {
   }
 
   static get defaultAdapter() {
+    // tslint:disable:object-literal-sort-keys
     return {
-      hasClass: (/* className: string */) => /* boolean */ false,
-      addClass: (/* className: string */) => {},
-      removeClass: (/* className: string */) => {},
-      registerScrollHandler: (/* handler: EventListener */) => {},
-      deregisterScrollHandler: (/* handler: EventListener */) => {},
-      registerResizeHandler: (/* handler: EventListener */) => {},
-      deregisterResizeHandler: (/* handler: EventListener */) => {},
-      getViewportWidth: () => /* number */ 0,
-      getViewportScrollY: () => /* number */ 0,
-      getOffsetHeight: () => /* number */ 0,
-      getFirstRowElementOffsetHeight: () => /* number */ 0,
-      notifyChange: (/* evtData: {flexibleExpansionRatio: number} */) => {},
-      setStyle: (/* property: string, value: string */) => {},
-      setStyleForTitleElement: (/* property: string, value: string */) => {},
-      setStyleForFlexibleRowElement: (/* property: string, value: string */) => {},
-      setStyleForFixedAdjustElement: (/* property: string, value: string */) => {},
+      hasClass: () => false,
+      addClass: () => undefined,
+      removeClass: () => undefined,
+      registerScrollHandler: () => undefined,
+      deregisterScrollHandler: () => undefined,
+      registerResizeHandler: () => undefined,
+      deregisterResizeHandler: () => undefined,
+      getViewportWidth: () => 0,
+      getViewportScrollY: () => 0,
+      getOffsetHeight: () => 0,
+      getFirstRowElementOffsetHeight: () => 0,
+      notifyChange: () => undefined,
+      setStyle: () => undefined,
+      setStyleForTitleElement: () => undefined,
+      setStyleForFlexibleRowElement: () => undefined,
+      setStyleForFixedAdjustElement: () => undefined,
     };
+    // tslint:enable:object-literal-sort-keys
   }
 
-  constructor(adapter) {
-    super(Object.assign(MDCToolbarFoundation.defaultAdapter, adapter));
+  private checkRowHeightFrame_ = 0;
+  private scrollFrame_ = 0;
+  private executedLastChange_ = false;
+  private isFixed_ = false;
+  private isFixedLastRow_ = false;
+  private hasFlexibleFirstRow_ = false;
+  private useFlexDefaultBehavior_ = false;
+  private calculations_: Calculations = {
+    flexibleExpansionHeight: 0,
+    flexibleExpansionRatio: 0,
+    maxTranslateYDistance: 0,
+    maxTranslateYRatio: 0,
+    scrollThreshold: 0,
+    scrollThresholdRatio: 0,
+    toolbarHeight: 0,
+    toolbarRatio: 0,
+    toolbarRowHeight: 0,
+  };
+
+  private readonly resizeHandler_: SpecificEventListener<'resize'>;
+  private readonly scrollHandler_: SpecificEventListener<'scroll'>;
+
+  constructor(adapter?: Partial<MDCToolbarAdapter>) {
+    super({...MDCToolbarFoundation.defaultAdapter, ...adapter});
+
     this.resizeHandler_ = () => this.checkRowHeight_();
```

Comment:
I think these can actually be defined outside the constructor

Reply:
Done